tst/regression/utils/test_case.py

```diff
@@ -104,6 +107,19 @@ def __init__(self,run_test_path,**kwargs):
         self.parameters.test_path = test_path
         self.parameters.mpi_cmd = mpi_executable
         self.parameters.mpi_opts = kwargs.pop('mpirun_opts')
+       
+        #if "-np " in self.parameters.mpi_opts:
+        print("MPI options %s" % self.parameters.mpi_opts)
+        try:
+          index = self.parameters.mpi_opts.index("-np")
+          self.parameters.num_ranks = int(self.parameters.mpi_opts[index+1])
+        except ValueError:
+          pass
+        try:
+          index = self.parameters.mpi_opts.index("-n")
+          self.parameters.num_ranks = int(self.parameters.mpi_opts[index+1])
+        except:
+          pass
```

Comment:
Using `try`-`except` logic feels a little unsafe to me here. I don't like that the `index` is behind the try along with the typecast. Also I don't love the same code being repeated twice. 

Also this implicitly sets `-n` to override `-np`. We should complain or warn or something if that's happening.

What about the following?
```suggestion
        argstrings = ['-np','-n']
        if len(set(argstrings) & set(self.parameters.mpi_opts)) > 1:
          print('Warning! You have set both "-n" and "-np" in your MPI options.")
        for s in argstrings:
          index = self.parameters.mpi_opts.index(s)
          if index < len(self.parameters.mpi_opts) - 1:
            try:
              self.parameters.num_ranks = int(self.parameters.mpi_opts[index+1])
            except ValueError:
              pass
```
Or am I missing something for why this wouldn't work?

Reply:
Yes, this would be an improvement.